src/Controller/Controller.php

```diff
@@ -757,12 +758,79 @@ public function render(?string $template = null, ?string $layout = null): Respon
         if ($builder->getTemplate() === null) {
             $builder->setTemplate($this->request->getParam('action'));
         }
+        $viewClass = $this->chooseViewClass();
+        $view = $this->createView($viewClass);
 
-        $view = $this->createView();
         $contents = $view->render();
-        $this->setResponse($view->getResponse()->withStringBody($contents));
+        $response = $view->getResponse()->withStringBody($contents);
 
-        return $this->response;
+        $viewContentType = $view->getContentType();
+        $responseType = $response->getHeaderLine('Content-Type');
+        if ($viewContentType && ($responseType === '' || substr($responseType, 0, 9) === 'text/html')) {
+            $response = $response->withHeader('Content-Type', $viewContentType);
```

Comment:
IMO it should be the view class' responsibility to set the proper content type. Currently `SerializedView` already do so.

Reply:
> IMO it should be the view class' responsibility to set the proper content type. Currently SerializedView already do so.

Where to put that logic though? We could put it the `initialize()` hook or `__construct()`.

Reply:
In `initialize()` sounds good.

Reply:
Done now. I also removed the get prefix on getContentType as I don't think it adds anything as there is no set again.